### PR TITLE
Add release GA workflow with goreleaser

### PR DESCRIPTION
**Pull request by nikimanoledaki**
Addresses https://github.com/grafana/cloudcost-exporter/issues/18

* Initialize goreleaser and its configuration
* Add a release GA workflow to create new releases when there are new tags
* Update docs about the release process

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    name_template: >-
+      {{ .ProjectName }}_
+      {{- title .Os }}_
+      {{- if eq .Arch "amd64" }}x86_64
+      {{- else }}{{ .Arch }}{{ end }}
+      {{- if .Arm }}v{{ .Arm }}{{ end }}
+checksum:
+  name_template: 'checksums.txt'
+changelog:
+  sort: asc
```
Comment (nikimanoledaki) [inline .goreleaser.yaml:27]: @Pokom we can add some rules here to exclude `chore` or `docs` if you want. My thinking was that we could leave all PR titles for now and iterate later if we want to make the release notes leaner.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+permissions:
+  contents: read
+
+jobs:
+  goreleaser:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
```
Comment (spinillos) [inline .github/workflows/release.yml:16]: For security purposes, we should set the SHA instead the version: [docs](https://enghub.grafana-ops.net/docs/default/component/deployment-tools/platform/continuous-integration/#pin-actions-to-a-full-length-commit-sha). Its something that we need to update in other repos 😅. Here you are an [example](https://github.com/grafana/deployment_tools/blob/master/.github/actions/kube-manifests-exporter/conftest-full/action.yml#L34).

Comment (Pokom) [reply .github/workflows/release.yml:16]: TIL! Thanks for sharing that @spinillos. I agree, let's pin to a specific version here and we can clean up other instances in another PR

File: .goreleaser.yaml (hunk 1 above)
Comment (Pokom) [reply .goreleaser.yaml:27]: Agreed on `docs` and think that would be fine to add here, assuming it's a simple check. `chore` may be related to code changes, so I would hold off on excluding `chore` PR's

File: .github/workflows/release.yml (hunk 2 above)
Comment (nikimanoledaki) [reply .github/workflows/release.yml]: I didn't know that, thank you! Changed it for the new actions, did the same for other actions that we use, and updated the docs :)

File: .goreleaser.yaml (hunk 1 above)
Comment (nikimanoledaki) [reply .goreleaser.yaml:27]: added exclusions for docs and specifically chore(deps) :)

File: docs/contribute/releases.md [hunk 3]
```diff
@@ -1,19 +1,45 @@
...
 To cut a release, clone `cloudcost-exporter` locally and pull latest from main.
-Determine if the next release is a major, minor, or hotfix. 
+Determine if the next release is a major, minor, or hotfix.
 Then increment the relevant version label.
 
 For instance, let's say we're on `v0.2.2` and determined the next release is a minor change.
-The next version would then be `v0.3.0`. 
-Execute the follow command and generate the tag and push it:
+The next version would then be `v0.3.0`.
+Execute the follow command to generate the tag and push it:
```
Comment (Pokom) [inline docs/contribute/releases.md:17]:
```suggestion
Execute the following command to generate the tag and push it:
```

**Review by Pokom — APPROVED** [review]
One very minor suggestion, otherwise 👍🏻

**Pull request merged**